framework/auth/campaigns.py

```diff
@@ -1,21 +1,27 @@
+from datetime import datetime, timedelta
 import furl
+import logging
 
 from modularodm import Q
 from modularodm.exceptions import NoResultsFound, QueryException, ImproperConfigurationError
 
 from website import mails
 from website.models import PreprintProvider
-from website.settings import DOMAIN
+from website.settings import DOMAIN, CAMPAIGN_REFRESH_THRESHOLD
 
 
 CAMPAIGNS = None
+CAMPAIGNS_LAST_REFRESHED = datetime.utcnow()
 
 
 def get_campaigns():
 
     global CAMPAIGNS
+    global CAMPAIGNS_LAST_REFRESHED
 
-    if not CAMPAIGNS:
+    logger = logging.getLogger(__name__)
+
+    if not CAMPAIGNS or CAMPAIGNS_LAST_REFRESHED + timedelta(seconds=CAMPAIGN_REFRESH_THRESHOLD) < datetime.utcnow():
```

Comment:
This is probably a good use for throttle_period_expired() - it's a simple function, but it's tested.